services/ui/ws/display.cc

```diff
@@ -77,14 +77,25 @@ Display::~Display() {
 }
 
 void Display::Init(const display::ViewportMetrics& metrics,
-                   std::unique_ptr<DisplayBinding> binding) {
+                   std::unique_ptr<DisplayBinding> binding,
+                   const ServerWindow* parent_window) {
   binding_ = std::move(binding);
   display_manager()->AddDisplay(this);
 
   CreateRootWindow(metrics.bounds_in_pixels.size());
 
-  platform_display_ = PlatformDisplay::Create(
-      root_.get(), metrics, window_server_->GetThreadedImageCursorsFactory());
+  WindowManagerDisplayRoot* display_root =
```

Comment:
one suggestion to make the logic tied to the parent_window parameter:

PlatformDisplay* parent_platform_display = nullptr;
if (parent_window) {
  WindowManagerDisplayRoot* parent_display_root = display_manager()->GetWindowManagerDisplayRoot(parent_window);
  DCHECK(parent_display_root);
  parent_platform_display = parent_display_root->display()->platform_display();
  DCHECK(parent_platform_display);
}